features/enabled_error_types.feature

```diff
@@ -1,144 +1,52 @@
-    #             "name": "P39disabled_cxx_reporting_kaboom_exception"
-    #           }
-    #         }
-    #       },
-    #       "exceptions": [
-    #         {
-    #           "message": "",
-    #           "errorClass": "P39disabled_cxx_reporting_kaboom_exception",
-    #           "stacktrace": [
-    #           ],
-    #           "type": "cocoa"
-    #         }
-    #       ],
-
-    When I crash the app using "EnabledErrorTypesCxxScenario"
-    And I relaunch the app
-    And I wait for 5 seconds
-    And I wait for 1 requests
-    And the request 0 is valid for the error reporting API
-    # Not a c++ exception
-    And the payload field "events.0.exceptions.0.errorClass" equals "SIGABRT" for request 0
-    And the payload field "events.0.metaData.error.type" equals "signal" for request 0
-
-# Typical Mach event:
-#
-# {
-#   "apiKey": "a35a2a72bd230ac0aa0f52715bbdc6aa",
-#   "payloadVersion": "4.0",
-#   "events": [
-#     {
-#       "metaData": {
-#         "error": {
-#           "address": 0,
-#           "mach": {
-#             "code": 0,
-#             "exception_name": "EXC_BAD_ACCESS",
-#             "subcode": 8,
-#             "exception": 1
-#           },
-#           "type": "mach"
-#         }
-#       },
-#       "exceptions": [
-#         {
-#           "message": "Attempted to dereference null pointer.",
-#           "errorClass": "EXC_BAD_ACCESS",
-#           "stacktrace": ...
+    When I run "EnabledErrorTypesCxxScenario" and relaunch the app
+    And I configure Bugsnag for "EnabledErrorTypesCxxScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
+    And the event "unhandled" is false
+    And the payload field "events.0.exceptions.0.message" equals "EnabledErrorTypesCxxScenario - Handled"
 
 Scenario: Mach Crash Reporting is disabled
-    When I crash the app using "DisableMachExceptionScenario"
+    When I run "DisableMachExceptionScenario"
     And I relaunch the app
-    And I wait for 1 requests
-    And the request 0 is valid for the error reporting API
-
-    # Not a Mach exception
-    # The Mach exception gets trampolined to a SEGV signal?
-    And the payload field "events.0.exceptions.0.errorClass" equals "SIGSEGV" for request 0
-    And the payload field "events.0.metaData.error.type" equals "signal" for request 0
+    And I configure Bugsnag for "DisableMachExceptionScenario"
+    And I wait to receive a request
+    Then the request is valid for the error reporting API version "4.0" for the "iOS Bugsnag Notifier" notifier
+    And the event "unhandled" is false
+    And the payload field "events.0.exceptions.0.message" equals "DisableMachExceptionScenario - Handled"
 
 Scenario: Signals Crash Reporting is disabled
-    When I crash the app using "DisableSignalsExceptionScenario"
-    And I relaunch the app
+    When I run "DisableSignalsExceptionScenario" and relaunch the app
```

Comment:
Would be great if we could do away with tests with no positive outcome as part of the move.  Can we test this in another way?  Can we move it into another test and test positive/negative outcomes in the same scenario at least?  Maybe not an easy answer and perhaps something to revisit.